apps/web/components/connected/UmbodsmadurSkuldara/UmsCostOfLivingCalculator.tsx

```diff
@@ -0,0 +1,580 @@
+      }
+    })
+
+    return total
+  }
+  const selectValue = (
+    value: string | undefined,
+  ): PropsValue<Option<unknown>> | undefined => {
+    for (const group of costOfLivingOptions) {
+      for (const opt of group.options) {
+        if (opt.value === value) {
+          return opt
+        }
+      }
+    }
+    return undefined
+  }
+
+  return (
+    <Box>
+      <Box
+        background="blue100"
+        paddingY={[3, 3, 6]}
+        paddingX={[3, 3, 3, 3, 15]}
+        marginBottom={5}
+      >
+        {dataState === 'loading' && (
+          <Box
+            display="flex"
+            marginTop={4}
+            marginBottom={20}
+            justifyContent="center"
+          >
+            <LoadingDots />
+          </Box>
+        )}
+        {dataState === 'error' && (
+          <AlertMessage
+            title={n('errorTitle', 'Villa')}
+            message={n(
+              'errorMessage',
+              activeLocale === 'is'
+                ? 'Ekki tókst að sækja gögn fyrir framfærslu reiknivél.'
+                : 'Failed to retieve data for cost of living calculator',
+            )}
+            type="error"
+          />
+        )}
+        {dataState === 'loaded' && (
+          <Stack space={5}>
+            <Stack space={2}>
+              <Text variant="h2">
+                {n(
+                  'costOfLivingCalculatorTitle',
+                  activeLocale === 'is' ? 'Fjölskyldustærð' : 'Family size',
+                )}
+              </Text>
+              <Controller
+                control={control}
+                name="familySize"
+                defaultValue={'1+0'}
+                render={({ field: { onChange, value } }) => (
+                  <Select
+                    label={n(
+                      'familySizeLabel',
+                      activeLocale === 'is' ? 'Fjölskyldustærð' : 'Family size',
+                    )}
+                    options={costOfLivingOptions}
+                    value={selectValue(value)}
+                    size="sm"
+                    isSearchable={false}
+                    onChange={(option) => {
+                      onChange(option?.value)
+                    }}
+                  />
+                )}
+              />
+            </Stack>
+            <Stack space={3}>
+              <Text variant="h3">
+                {n(
+                  'preCalculatedTitle',
+                  activeLocale === 'is'
+                    ? 'Framfærsluviðmið umboðsmanns skuldara'
+                    : 'Subsistence criteria',
+                )}
+              </Text>
+
+              <CalculatedFieldWrapper
+                title={n(
+                  'foodLabel',
+                  activeLocale === 'is'
+                    ? 'Matur, hreinlætisvörur og heimilisbúnaður'
+                    : 'Food, hygiene products and household equipment',
+                )}
+                value={getSelectedValues(familySize)?.food}
```

Comment:
Instead of calling getSelectedValues over and over again, why not just create a variable above and call it once?